Version Bump + Changelogs for 5.0.0 (#2976)
moderntribe/the-events-calendar: readme.txt

== Changelog ==

= [5.0.0] TBD =

> * Feature - Month, Day and List views HTML, CSS and JavaScript were completely refactored.

Reviewer: This is sort of accurate. What we did was add _new_ Month, Day, and List views with new HTML, CSS, and JavaScript.

```suggestion
* Feature - Created new Month, Day, and List views with all new HTML, CSS, and JavaScript.
```

> * Feature - Entirely new templating engine to power the Views, one that will pass all pieces of template more filters and actions. [See More]

Reviewer: ```suggestion
* Feature - New templating engine to power the new Views. [See More]
```

> * Feature - Views initialization and instance contains a lot more filters and actions to allow improved customization and even addition of more views. (See all filters below)

Reviewer: ```suggestion
* Feature - Views initialization and instance includes more filters and actions to allow improved extensibility and customization. (See all filters below)
```

* Feature - Views manager and bootstraping classes were created to allow more control over where and when views will be initialized. (See filters related to those below)

> * Feature - Increased granularity on the templates to help long term maintenance for theme customization on any of the Views refactored code. (See list of files now available below)

Reviewer: ```suggestion
* Feature - Reduced maintenance burden of customizations by leveraging atomic templates for the new views. (See list of files now available below)
```

* Feature - CSS uses a Block Element Modifier (BEM) methodology to improve reusablility of components. [See More]

> * Feature - All of the Views share one JavaScript Manager, which contains mutiple jQuery actions, allowing a lot more customizations points. [See More]

Reviewer: ```suggestion
* Feature - New Views share one JavaScript Manager, which contains multiple jQuery actions and customization points. [See More]
```

> * Feature - Views navigation relies on the WP Rest API which improves overall AJAX reliability and possibility of caching.

Reviewer: ```suggestion
* Feature - View navigation relies on the WP Rest API, improving reliability and caching opportunities.
```

> * Feature - SEO improved for Events permalinks which are more consistent across navigation, making use of `WP_Rewrite` to map params to their matching URL.

Reviewer: ```suggestion
* Feature - Event permalinks are now more consistent across navigation using `WP_Rewrite` to map params to their matching URL, which improves SEO.
```

> * Feature - Event individual filtering for data, used across all refactored views: `tribe_get_event_before`, `tribe_get_event`, `tribe_get_event_after`, `tribe_events_organizers_fetch_callback`, `tribe_events_venues_fetch_callback`

Reviewer: ```suggestion
* Feature - Event data filtering used across all new views: `tribe_get_event_before`, `tribe_get_event`, `tribe_get_event_after`, `tribe_events_organizers_fetch_callback`, `tribe_events_venues_fetch_callback`
```

> * Feature - Venue individual filtering for data, used across all refactored views: `tribe_get_venue_object_before`, `tribe_get_venue_object`

Reviewer: ```suggestion
* Feature - Venue data filtering used across all new views: `tribe_get_venue_object_before`, `tribe_get_venue_object`
```

> * Tweak - Custom HTML for After and Before modified on the Events > Settings page, now will no longer use the Theme's CSS.

Reviewer: ```suggestion
* Tweak - Custom Before and After HTML on the Events > Settings page will now prefer event view styling over the Theme's CSS.
```

* Tweak - Lowered the reliance on the global `$wp_query` to create more consistent output and faster maintenance. [See More]

> * Tweak - Views refactored templates will be looking for a new folder in your theme to avoid conflicting with old The Events Calendar theme customizations [See More]

Reviewer: ```suggestion
* Tweak - View overrides for the new views will look within a new folder in your theme to avoid conflicting with old The Events Calendar theme customizations [See More]
```
